pyresample/geometry.py

```diff
@@ -314,6 +324,20 @@ def get_bbox_lonlats(self, force_clockwise: bool = True) -> tuple:
             lons, lats = self._reverse_boundaries(lons, lats)
         return lons, lats
 
+    def _get_bbox_slices(self, frequency):
+        height, width = self.shape
+        if frequency is None:
+            line_num = height
+            col_num = width
+        else:
+            line_num = frequency
+            col_num = frequency
+        s1_slice = (0, np.linspace(0, width - 1, col_num, dtype=int))
+        s2_slice = (np.linspace(0, height - 1, line_num, dtype=int), -1)
+        s3_slice = (-1, np.linspace(width - 1, 0, col_num, dtype=int))
+        s4_slice = (np.linspace(height - 1, 0, line_num, dtype=int), 0)
```

Comment:
This replaces a slice with step because we need to make sure the last element is kept. This can lead to uneven steps that would require some ugly hacking to get in slice syntax. However, I'm not sure I see how a slice with steps would be faster than a list of indices, since I assume a slice with steps is converted to a list of indices. Or do you have some inside info?

Reply:
A list of indices is considered fancy indexing isn't it? Fancy indexing performs worse if I remember correctly because dask can't make the same optimizations. This may not be true in modern dask, but I know we had to be careful about the type of indexing we were using in the past.

Reply:
Looks good nowadays... or should I make a bigger array?
```python
In [10]: z = da.zeros((1000000, 1000000))

In [11]: %time z[slice(0, None, 100), :]
CPU times: user 167 ms, sys: 13.9 ms, total: 180 ms
Wall time: 180 ms
Out[11]: dask.array<getitem, shape=(10000, 1000000), dtype=float64, chunksize=(40, 4000), chunktype=numpy.ndarray>

In [12]: %time z[np.linspace(0, 1000000 - 1, 10000, dtype=int), :]
CPU times: user 50.8 ms, sys: 10.1 ms, total: 60.9 ms
Wall time: 60.3 ms
```

Reply:
with 10x10 million values, indexing with slice takes 14.5 seconds, with linspace takes 12 seconds...

Reply:
I did another test where I made the step size larger so it didn't include every chunk and then used `.visualize` to see what the dask graph looked like. They seem to be equivalent. I'm not sure why, but when not every chunk is used the dask graph still shows the unused chunks:

![image](https://user-images.githubusercontent.com/1828519/171187581-7637450a-5a34-424e-a307-c5188fb4d99d.png)

And yeah it looks like the timing is pretty equal:

```
In [40]: %timeit x = z[np.linspace(0, 100000 - 1, 10, dtype=np.int64), :].compute()
40.9 ms ± 406 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [41]: %timeit x = z[slice(0, None, 10000), :].compute()
40.1 ms ± 385 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

Looks good to me.